yt_dlp/extractor/common.py

```diff
@@ -860,8 +862,30 @@ def _request_webpage(self, url_or_request, video_id, note=None, errnote=None, fa
             headers = (headers or {}).copy()
             headers.setdefault('X-Forwarded-For', self._x_forwarded_for_ip)
 
+        extensions, requested_targets = {}, []
+        if impersonate is True:
+            requested_targets.append(ImpersonateTarget())
+        elif isinstance(impersonate, ImpersonateTarget):
+            requested_targets.append(impersonate)
+        elif isinstance(impersonate, (str, list, tuple)):
+            requested_targets = tuple(map(ImpersonateTarget.from_str, variadic(impersonate)))
```

Comment:
* If none of these match, the argument is silently ignored? We should error instead imo.
* Shouldn't it also support `list[ImpersonateTarget]` for consistency?

```suggestion
        if impersonate is True:
            impersonate = ['']
        requested_targets = [
            x if isinstance(x, ImpersonateTarget) else ImpersonateTarget.from_str(x)
            for x in variadic(impersonate)
        ]
        extensions = {}
```


Also, do we need `variadic`, or only support list? I'm not sure

Reply:
your suggestion breaks if `impersonate` is `None` (default) or `False`

> If none of these match, the argument is silently ignored? We should error instead imo.

`False` and `None` should be silently ignored, yes. I suppose we could raise if it's otherwise truthy, but I don't see how to do that within the above suggestion

> only support list?

I originally only wanted to support strings and `True`; IMO it's unintuitive from the perspective of an extractor author to have to pass a list

Reply:
> Shouldn't it also support `list[ImpersonateTarget]` for consistency?

If we're passing `ImpersonateTarget` at all, then there has already been some logic in the extractor to arrive at what is being passed and a list of `ImpersonateTarget` instances seems like overkill to me. But I guess we could try to do it if you think so

Reply:
> your suggestion breaks if `impersonate` is `None` (default) or `False`

> I suppose we could raise if it's otherwise truthy, but I don't see how to do that within the above suggestion

`requested_targets = [...] if impersonate else []`?

edited suggestion.